sink/sink.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+    "bufio"
+    "fmt"
+    "github.com/prometheus/client_golang/prometheus"
+    "github.com/prometheus/client_golang/prometheus/promauto"
+    "github.com/prometheus/client_golang/prometheus/promhttp"
+    "net"
+    "net/http"
+    "strconv"
+    "strings"
+    "time"
+)
+
+var (
+        metricsInjectorReceived = promauto.NewCounter(prometheus.CounterOpts{
+                Name: "sink_metrics_injector_received_count",
+                Help: "The total number of metrics received from injector",
+        })
+        metricsCarbonReceived = promauto.NewCounter(prometheus.CounterOpts{
+                Name: "sink_metrics_carbon_received_count",
+                Help: "The total number of metrics received from carbon relay",
+        })
+        metricsLatency = promauto.NewHistogram(prometheus.HistogramOpts{
+                Name:    "sink_metrics_latency",
+                Help:    "Histogram latency metrics", 
+		Buckets: prometheus.ExponentialBuckets(10000000,10,3),
+        })
+)
+
+func getLatency(epochTimeStamp int64) int64 {
+  currentEpochTimeInSec := time.Now().UnixNano()
+  latency := currentEpochTimeInSec - epochTimeStamp
+  return latency
+}
+
+func isMetricFromCarbonRelayNg(metricPath string) bool {
+   return strings.HasPrefix(metricPath,"service_is_carbon-relay-ng")
+}
+
+func handleMetricMessage(message string) {
+  metricMessage := strings.Split(message," ")
```

Comment:
I find the use of `message` and `metricMessage` confusing: it it hard to determine what is the link between them and what makes them different without looking at the variable definition.
Maybe something like `message` and `messageArray` would be more clear?

Another point: the Carbon documentation uses `data` and not `message` about it. 
For consistency, what about using the same terminology as they do?